src/FSharp.Core/prim-types.fs

```diff
@@ -377,6 +377,12 @@ namespace Microsoft.FSharp.Core
     type NoCompilerInliningAttribute() =
         inherit Attribute()
 
+    [<AttributeUsage (AttributeTargets.Method, AllowMultiple=false)>]  
+    [<Sealed>]
+    type WarnOnWithoutNullArgumentAttribute(warningMessage:string) =
+        inherit Attribute()
+        member _.WarningMessage = warningMessage
+        
```

Comment:
Wonder if this will be useful outside the fslib.

Reply:
I have seen custom project-specific null handlers pretty much everywhere, so I believe yes.
Also, I did not like making this internal and consuming it from F.C.S, but it would be an option too.